src/app/users/users.component.html

```diff
@@ -86,12 +86,9 @@
       <ng-container matColumnDef="name">
         <mat-header-cell *matHeaderCellDef mat-sort-header i18n>Name</mat-header-cell>
         <mat-cell *matCellDef="let element">
-          <span *ngIf="element.doc.firstName || element.doc.lastName; else origName">
-            {{element.doc.firstName}} {{element.doc.middleName}} {{element.doc.lastName}}
+          <span>
+            {{ element.doc.firstName || element.doc.lastName ? element.fullName : element.doc.name }}

```

Comment:
This conditional should be the value for `fullName` so the search works when `element.doc.name` is used.

Reply:
Hey Paul, I have a question. Do you mean something like this ... `{{ element.fullName ? element.fullName : element.doc.name }}` ?